contracts/Airdrop.sol

```diff
@@ -23,42 +27,42 @@ contract Airdrop is Ownable {
     event AirdropClaim(uint256 indexed airdropID, address indexed claimer, uint256 amount);
     event AirdropInit(uint256 indexed airdropID, uint256 indexed initAt, uint256 amount);
     event AirdropTokenRevised(address indexed oldToken, address indexed newToken);
+    event ThresholdUpdated(uint256 indexed oldThreshold, uint256 indexed newThreshold);
+    event ThresholdReached(uint256 indexed balance, uint256 threshold, string info);
 
-    constructor(IERC20 _airdropToken, address _owner) Ownable(_owner) {
+    constructor(IERC20 _airdropToken, address _owner, uint256 _thresholdTokenAmount) Ownable(_owner) {
         airdropToken = _airdropToken;
+        thresholdBalance = _thresholdTokenAmount;
     }
 
-    /* 
-    @param _numberOfUsers number of users should be same as user addresses feeded in the merkle tree.
-    @param _amount amount should be constant for all users in merkle tree.
-    */
-
     function init(bytes32 _merkleRoot, uint256 _numberOfUsers, uint256 _amount)
         external
         onlyOwner
         returns (uint256 airdropID)
     {
         require(_merkleRoot != bytes32(0), "Airdrop: Merkle Root should not zero bytes");
         require(_numberOfUsers > 0, "Airdrop: number of users should not zero");
-        airDrops[airdropsCount] = AirdropInfos({
-            merkleRoot: _merkleRoot,
-            numberOfUsers: _numberOfUsers,
-            amount: _amount,
-            startedAt: block.timestamp
-        });
+        airDrops[airdropsCount] =
+            AirdropInfos({merkleRoot: _merkleRoot, numberOfUsers: _numberOfUsers, startedAt: block.timestamp});
         airdropID = airdropsCount;
         airdropsCount++;
         emit AirdropInit(airdropID, block.timestamp, _amount);
     }
 
     function revise(IERC20 _airdropToken) external onlyOwner {
-        address newToken = address(_airdropToken)
+        address newToken = address(_airdropToken);
         require(newToken != address(0), "Airdrop: Token should not zero address");
         address oldToken = address(airdropToken);
```

Comment:
Unnecessary code.
`require(_airdropToken != address(0), "Airdrop: Token should not zero address");`
`airdropToken = _airdropToken;`
emit event with _airdropToken only.

Reply:
Fixed in [fa35427](https://github.com/MisterDefender/Airdrop-ERC-20/pull/2/commits/fa354274c837d5e19adbbf4e81cd8a5444bb1c20)